src/controller/java/src/matter/controller/States.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package matter.controller
```

Comment:
should this maybe be in some separate package?

I think state management should sit at some layer above the raw controller functions.

Reply:
Looking at their usage currently, it seems we use this to wrap the java version of states.

I would claim that this is an implementation detail of java using nodestate, however that would not be required if we had direct jni calls, so we could not have these at all.

Reply:
moved Path.kt and States.kt to package matter.controller.model